### PR TITLE
Improve UT test coverage for iosxr

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### SUMMARY

Add missing uts for following iosxr modules: interfaces, l2_interfaces, l3_interfaces, lag_interfaces, acl_interfaces, lldp_global, lldp_interfaces, lacp, lacp_interfaces, banner

Coverage report: 69%

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

iosxr_interfaces,iosxr_l2_interfaces,iosxr_l3_interfaces,iosxr_acl_interfaces,iosxr_lacp_interfaces,iosxr_lag_interfaces,iosxr_lldp_interfaces,iosxr_lldp_global,iosxr_banner

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by ashwini-mhatre** [comment]
<div id="header" style="background: rgb(248, 248, 248); width: 1792px; border-bottom: 1px solid rgb(238, 238, 238); color: rgb(0, 0, 0); font-family: georgia, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="content" style="padding: 1em 3em;"><h1 style="margin: 0px; padding: 0px; border: 0px; outline: 0px; font-weight: inherit; font-style: inherit; font-size: 1.25em; font-family: inherit; vertical-align: baseline; display: inline-block;">Coverage report:<span> </span><span class="pc_cov">69%</span></h1>Show keyboard shortcuts<form id="filter_container" style="display: inline-block; float: right; margin: 0px 2em 0px 0px;"><input id="filter" type="text" value="" placeholder="filter..." style="width: 10em;"></form></div></div><div id="index" style="margin: 1em 0px 0px 3em; color: rgb(0, 0, 0); font-family: georgia, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Module | statements | missing | excluded | branches | partial | coverage
-- | -- | -- | -- | -- | -- | --
Total | 9548 | 2317 | 0 | 4075 | 533 | 69%
plugins/cliconf/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/cliconf/iosxr.py | 182 | 80 | 0 | 66 | 9 | 47%
plugins/module_utils/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/acl_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/acl_interfaces/acl_interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/acls/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/acls/acls.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/bgp_address_family/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/bgp_address_family/bgp_address_family.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/bgp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/bgp_global/bgp_global.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/bgp_neighbor_address_family/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/bgp_neighbor_address_family/bgp_neighbor_address_family.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/facts/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/facts/facts.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/interfaces/interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/l2_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/l2_interfaces/l2_interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/l3_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/l3_interfaces/l3_interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/lacp/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/lacp/lacp.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/lacp_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/lacp_interfaces/lacp_interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/lag_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/lag_interfaces/lag_interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/lldp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/lldp_global/lldp_global.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/lldp_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/lldp_interfaces/lldp_interfaces.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/logging_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/logging_global/logging_global.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/ospf_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/ospf_interfaces/ospf_interfaces.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/ospfv2/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/ospfv2/ospfv2.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/argspec/ospfv3/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/ospfv3/ospfv3.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/prefix_lists/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/prefix_lists/prefix_lists.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/argspec/static_routes/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/static_routes/static_routes.py | 7 | 1 | 0 | 0 | 0 | 86%
plugins/module_utils/network/iosxr/config/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/acl_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/acl_interfaces/acl_interfaces.py | 54 | 0 | 0 | 40 | 0 | 100%
plugins/module_utils/network/iosxr/config/acls/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/acls/acls.py | 219 | 34 | 0 | 146 | 21 | 79%
plugins/module_utils/network/iosxr/config/bgp_address_family/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/bgp_address_family/bgp_address_family.py | 85 | 5 | 0 | 52 | 2 | 93%
plugins/module_utils/network/iosxr/config/bgp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/bgp_global/bgp_global.py | 125 | 28 | 0 | 76 | 10 | 73%
plugins/module_utils/network/iosxr/config/bgp_neighbor_address_family/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py | 103 | 9 | 0 | 62 | 5 | 88%
plugins/module_utils/network/iosxr/config/interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/interfaces/interfaces.py | 164 | 12 | 0 | 92 | 10 | 89%
plugins/module_utils/network/iosxr/config/l2_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/l2_interfaces/l2_interfaces.py | 219 | 46 | 0 | 136 | 22 | 72%
plugins/module_utils/network/iosxr/config/l3_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/l3_interfaces/l3_interfaces.py | 209 | 23 | 0 | 122 | 18 | 83%
plugins/module_utils/network/iosxr/config/lacp/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/lacp/lacp.py | 89 | 25 | 0 | 45 | 15 | 60%
plugins/module_utils/network/iosxr/config/lacp_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/lacp_interfaces/lacp_interfaces.py | 135 | 8 | 0 | 78 | 8 | 91%
plugins/module_utils/network/iosxr/config/lag_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/lag_interfaces/lag_interfaces.py | 207 | 16 | 0 | 116 | 19 | 87%
plugins/module_utils/network/iosxr/config/lldp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/lldp_global/lldp_global.py | 105 | 3 | 0 | 58 | 10 | 92%
plugins/module_utils/network/iosxr/config/lldp_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/lldp_interfaces/lldp_interfaces.py | 127 | 7 | 0 | 72 | 10 | 90%
plugins/module_utils/network/iosxr/config/logging_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/logging_global/logging_global.py | 85 | 1 | 0 | 73 | 3 | 97%
plugins/module_utils/network/iosxr/config/ospf_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/ospf_interfaces/ospf_interfaces.py | 109 | 7 | 0 | 76 | 3 | 94%
plugins/module_utils/network/iosxr/config/ospfv2/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/ospfv2/ospfv2.py | 87 | 8 | 0 | 62 | 7 | 87%
plugins/module_utils/network/iosxr/config/ospfv3/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/ospfv3/ospfv3.py | 86 | 8 | 0 | 62 | 7 | 87%
plugins/module_utils/network/iosxr/config/prefix_lists/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/prefix_lists/prefix_lists.py | 58 | 0 | 0 | 47 | 2 | 98%
plugins/module_utils/network/iosxr/config/static_routes/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/static_routes/static_routes.py | 212 | 23 | 0 | 138 | 17 | 86%
plugins/module_utils/network/iosxr/facts/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/acl_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/acl_interfaces/acl_interfaces.py | 35 | 3 | 0 | 16 | 2 | 90%
plugins/module_utils/network/iosxr/facts/acls/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/acls/acls.py | 143 | 48 | 0 | 68 | 11 | 60%
plugins/module_utils/network/iosxr/facts/bgp_address_family/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/bgp_address_family/bgp_address_family.py | 42 | 3 | 0 | 10 | 3 | 88%
plugins/module_utils/network/iosxr/facts/bgp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/bgp_global/bgp_global.py | 67 | 5 | 0 | 34 | 8 | 87%
plugins/module_utils/network/iosxr/facts/bgp_neighbor_address_family/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/bgp_neighbor_address_family/bgp_neighbor_address_family.py | 47 | 3 | 0 | 18 | 4 | 89%
plugins/module_utils/network/iosxr/facts/facts.py | 37 | 0 | 0 | 4 | 2 | 95%
plugins/module_utils/network/iosxr/facts/interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/interfaces/interfaces.py | 59 | 7 | 0 | 26 | 6 | 82%
plugins/module_utils/network/iosxr/facts/l2_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/l2_interfaces/l2_interfaces.py | 90 | 16 | 0 | 50 | 13 | 75%
plugins/module_utils/network/iosxr/facts/l3_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/l3_interfaces/l3_interfaces.py | 68 | 7 | 0 | 28 | 5 | 85%
plugins/module_utils/network/iosxr/facts/lacp/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/lacp/lacp.py | 39 | 11 | 0 | 10 | 4 | 65%
plugins/module_utils/network/iosxr/facts/lacp_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/lacp_interfaces/lacp_interfaces.py | 50 | 3 | 0 | 20 | 2 | 93%
plugins/module_utils/network/iosxr/facts/lag_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/lag_interfaces/lag_interfaces.py | 62 | 4 | 0 | 26 | 5 | 90%
plugins/module_utils/network/iosxr/facts/legacy/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/legacy/base.py | 180 | 33 | 0 | 62 | 15 | 74%
plugins/module_utils/network/iosxr/facts/lldp_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/lldp_global/lldp_global.py | 44 | 3 | 0 | 18 | 3 | 90%
plugins/module_utils/network/iosxr/facts/lldp_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/lldp_interfaces/lldp_interfaces.py | 49 | 3 | 0 | 22 | 2 | 93%
plugins/module_utils/network/iosxr/facts/logging_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/logging_global/logging_global.py | 34 | 0 | 0 | 10 | 1 | 98%
plugins/module_utils/network/iosxr/facts/ospf_interfaces/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/ospf_interfaces/ospf_interfaces.py | 79 | 7 | 0 | 34 | 6 | 88%
plugins/module_utils/network/iosxr/facts/ospfv2/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/ospfv2/ospfv2.py | 90 | 13 | 0 | 50 | 15 | 79%
plugins/module_utils/network/iosxr/facts/ospfv3/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/ospfv3/ospfv3.py | 89 | 13 | 0 | 50 | 15 | 78%
plugins/module_utils/network/iosxr/facts/prefix_lists/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/prefix_lists/prefix_lists.py | 26 | 1 | 0 | 6 | 0 | 97%
plugins/module_utils/network/iosxr/facts/static_routes/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/static_routes/static_routes.py | 98 | 3 | 0 | 44 | 4 | 95%
plugins/module_utils/network/iosxr/iosxr.py | 308 | 224 | 0 | 146 | 5 | 21%
plugins/module_utils/network/iosxr/providers/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/providers/providers.py | 90 | 64 | 0 | 28 | 0 | 22%
plugins/module_utils/network/iosxr/rm_templates/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/rm_templates/acl_interfaces.py | 8 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/rm_templates/bgp_address_family.py | 196 | 139 | 0 | 136 | 15 | 22%
plugins/module_utils/network/iosxr/rm_templates/bgp_global.py | 16 | 7 | 0 | 4 | 0 | 45%
plugins/module_utils/network/iosxr/rm_templates/bgp_neighbor_address_family.py | 97 | 26 | 0 | 68 | 26 | 59%
plugins/module_utils/network/iosxr/rm_templates/logging_global.py | 80 | 1 | 0 | 46 | 16 | 87%
plugins/module_utils/network/iosxr/rm_templates/ospf_interfaces.py | 93 | 43 | 0 | 24 | 2 | 44%
plugins/module_utils/network/iosxr/rm_templates/ospfv2.py | 390 | 342 | 0 | 286 | 0 | 7%
plugins/module_utils/network/iosxr/rm_templates/ospfv3.py | 398 | 348 | 0 | 294 | 0 | 7%
plugins/module_utils/network/iosxr/rm_templates/prefix_lists.py | 8 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/utils/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/utils/utils.py | 269 | 64 | 0 | 193 | 40 | 72%
plugins/modules/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/modules/iosxr_acl_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_acls.py | 16 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_banner.py | 111 | 34 | 0 | 40 | 10 | 60%
plugins/modules/iosxr_bgp_address_family.py | 14 | 0 | 0 | 2 | 1 | 94%
plugins/modules/iosxr_bgp_global.py | 14 | 0 | 0 | 2 | 1 | 94%
plugins/modules/iosxr_bgp_neighbor_address_family.py | 14 | 1 | 0 | 2 | 1 | 88%
plugins/modules/iosxr_command.py | 57 | 2 | 0 | 22 | 1 | 96%
plugins/modules/iosxr_config.py | 119 | 20 | 0 | 52 | 10 | 78%
plugins/modules/iosxr_facts.py | 27 | 0 | 0 | 6 | 1 | 97%
plugins/modules/iosxr_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_l2_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_l3_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_lacp.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_lacp_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_lag_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_lldp_global.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_lldp_interfaces.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_logging_global.py | 15 | 1 | 0 | 2 | 1 | 88%
plugins/modules/iosxr_netconf.py | 82 | 2 | 0 | 45 | 8 | 91%
plugins/modules/iosxr_ospf_interfaces.py | 14 | 0 | 0 | 2 | 1 | 94%
plugins/modules/iosxr_ospfv2.py | 17 | 0 | 0 | 2 | 1 | 95%
plugins/modules/iosxr_ospfv3.py | 14 | 0 | 0 | 2 | 1 | 94%
plugins/modules/iosxr_prefix_lists.py | 14 | 1 | 0 | 2 | 1 | 88%
plugins/modules/iosxr_static_routes.py | 17 | 1 | 0 | 2 | 1 | 89%
plugins/modules/iosxr_system.py | 280 | 162 | 0 | 150 | 16 | 37%
plugins/modules/iosxr_user.py | 373 | 202 | 0 | 214 | 35 | 39%
tests/unit/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
tests/unit/compat/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
tests/unit/compat/mock.py | 52 | 45 | 0 | 20 | 1 | 11%
tests/unit/compat/unittest.py | 9 | 4 | 0 | 2 | 1 | 55%
tests/unit/modules/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/conftest.py | 20 | 12 | 0 | 10 | 0 | 27%
tests/unit/modules/network/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/iosxr_module.py | 46 | 0 | 0 | 8 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr.py | 44 | 9 | 0 | 8 | 1 | 73%
tests/unit/modules/network/iosxr/test_iosxr_acl_interfaces.py | 62 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_acls.py | 79 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_banner.py | 53 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_bgp_address_family.py | 73 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_bgp_global.py | 72 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_bgp_neighbor_address_family.py | 74 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_command.py | 65 | 0 | 0 | 2 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_config.py | 159 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_facts.py | 59 | 1 | 0 | 2 | 0 | 98%
tests/unit/modules/network/iosxr/test_iosxr_interfaces.py | 70 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_l2_interfaces.py | 79 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_l3_interfaces.py | 71 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_lacp.py | 36 | 3 | 0 | 0 | 0 | 92%
tests/unit/modules/network/iosxr/test_iosxr_lacp_interfaces.py | 71 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_lag_interfaces.py | 71 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_lldp_global.py | 63 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_lldp_interfaces.py | 71 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_logging_global.py | 71 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_netconf.py | 42 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_ospf_interfaces.py | 76 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_ospfv2.py | 73 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_ospfv3.py | 73 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_prefix_lists.py | 71 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_static_routes.py | 73 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_system.py | 55 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/network/iosxr/test_iosxr_user.py | 56 | 0 | 0 | 0 | 0 | 100%
tests/unit/modules/utils.py | 34 | 0 | 0 | 6 | 2 | 95%

</div><div id="footer" style="margin: 1em 3em; color: rgb(0, 0, 0); font-family: georgia, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="content" style="padding: 0px; font-size: 13.6px; font-family: verdana, sans-serif; color: rgb(102, 102, 102); font-style: italic;"><p style="margin: 0px; padding: 0px; border: 0px; outline: 0px; font-weight: inherit; font-style: inherit; font-size: 0.75em; font-family: inherit; vertical-align: baseline; line-height: 1.33333em;"><a class="nav" href="https://coverage.readthedocs.io" style="text-decoration: none; color: inherit;">coverage.py v4.5.4</a>, created at 2021-08-04 12:30</p></div></div>

File: changelogs/fragments/improve_test_coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
```
Comment (Qalthos) [inline changelogs/fragments/improve_test_coverage.yaml:2]: I'd argue this is `trivial`... test and coverage updates aren't generally something I think needs to be called out in a changelog.

File: plugins/module_utils/network/iosxr/facts/acl_interfaces/acl_interfaces.py [hunk 2]
```diff
@@ -44,6 +44,9 @@ def __init__(self, module, subspec="config", options="options"):
 
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
+    def get_config(self, connection):
+        return connection.get("show running-config interface")
```
Comment (Qalthos) [inline plugins/module_utils/network/iosxr/facts/acl_interfaces/acl_interfaces.py:48]: I'm assuming you're adding this function to make it easier to mock the result of `get_config`, but I am curious why you are calling `get` and not continuing to use `get_config` here

Comment (ashwini-mhatre) [reply plugins/module_utils/network/iosxr/facts/acl_interfaces/acl_interfaces.py:48]: yes

Comment (NilashishC) [reply plugins/module_utils/network/iosxr/facts/acl_interfaces/acl_interfaces.py:48]: +1 for changing this to `get_config`

File: plugins/module_utils/network/iosxr/facts/interfaces/interfaces.py [hunk 3]
```diff
@@ -44,6 +44,9 @@ def __init__(self, module, subspec="config", options="options"):
 
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
+    def get_config(self, connection):
+        return connection.get("show running-config interface")
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/facts/interfaces/interfaces.py:48]: Same as above.

File: plugins/module_utils/network/iosxr/facts/l2_interfaces/l2_interfaces.py [hunk 4]
```diff
@@ -49,6 +49,9 @@ def __init__(self, module, subspec="config", options="options"):
 
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
+    def get_config(self, connection):
+        return connection.get("show running-config interface")
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/facts/l2_interfaces/l2_interfaces.py:53]: Same as above.

File: plugins/module_utils/network/iosxr/facts/l3_interfaces/l3_interfaces.py [hunk 5]
```diff
@@ -46,6 +46,9 @@ def __init__(self, module, subspec="config", options="options"):
 
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
+    def get_config(self, connection):
+        return connection.get("show running-config interface")
```
Comment (NilashishC) [inline plugins/module_utils/network/iosxr/facts/l3_interfaces/l3_interfaces.py:50]: Same as above.

**Review by NilashishC — APPROVED** [review]

**Pull request merged**